embrace-android-sdk/src/main/java/io/embrace/android/embracesdk/comms/delivery/EmbraceCacheService.kt

```diff
@@ -186,43 +241,52 @@ internal class EmbraceCacheService(
         }
     }
 
-    override fun replaceSession(name: String, transformer: (SessionMessage) -> SessionMessage) {
+    override fun transformSession(name: String, transformer: (SessionMessage) -> SessionMessage) {
         findLock(name).write {
             try {
                 val sessionMessage = loadObject(name, SessionMessage::class.java) ?: return@write
-                val sessionFile = storageService.getFileForWrite(EMBRACE_PREFIX + name)
-                val oldSessionFile = storageService.getFileForWrite("${sessionFile.name}-old")
-                val newSessionFile = storageService.getFileForWrite("${sessionFile.name}-new")
                 val newMessage = transformer(sessionMessage)
-                serializer.toJson(newMessage, SessionMessage::class.java, newSessionFile.outputStream())
-                val oldSessionRenamed = sessionFile.renameTo(oldSessionFile)
-                val newSessionRenamed = newSessionFile.renameTo(sessionFile)
-                val oldSessionDeleted = oldSessionFile.delete()
-                if (!oldSessionRenamed || !newSessionRenamed || !oldSessionDeleted) {
-                    val failures = mutableListOf<String>()
-                    if (!oldSessionRenamed) {
-                        failures.add("old session file not renamed")
-                    }
-                    if (!newSessionRenamed) {
-                        failures.add("new session file not renamed")
-                    }
-                    if (!oldSessionDeleted) {
-                        failures.add("old session file not deleted")
-                    }
-                    logger.logError("Operation to replace the session failed: $failures")
-                }
+                writeSession(name, newMessage)
             } catch (ex: Exception) {
                 logger.logError("Failed to replace session object ", ex)
-                deleteFile(name)
             }
         }
     }
 
+    private fun replaceSessionFile(filenameToReplace: String, filenameOfReplacement: String): Boolean {
+        try {
+            val sessionFile = storageService.getFileForWrite(EMBRACE_PREFIX + filenameToReplace)
+            val newSessionFile = storageService.getFileForWrite(EMBRACE_PREFIX + filenameOfReplacement)
+            val oldSessionFile = storageService.getFileForWrite(sessionFile.name + OLD_COPY_SUFFIX)
+            if (sessionFile.exists()) {
+                sessionFile.renameTo(oldSessionFile)
+                oldSessionFile.deleteOnExit()
```

Comment:
Why `deleteOnExit` rather than `delete` here? Is that intentional?

Reply:
Actually yeah, deleting here as well is good. I refactored the code around this but didn't update this, so I'll do that.